File: Degree/Practical(IP)/P5Q1/Loan.php
<?php

class Loan {

    private $annualInterestRate;
    private $numberOfYears;
    private $loanAmount;

    public function __construct($annualInterestRate, $numberOfYears, $loanAmount) {
        $this->annualInterestRate = $annualInterestRate;
        $this->numberOfYears = $numberOfYears;
        $this->loanAmount = $loanAmount;
    }

    public function Loan($annualInterestRate = 2.5, $numberOfYears = 15, $loanAmount = 100000.0) {
        $this->annualInterestRate = $annualInterestRate;
        $this->numberOfYears = $numberOfYears;
        $this->loanAmount = $loanAmount;
    }

    public function getAnnualInterestRate() {
        return $this->annualInterestRate;
    }

    public function setAnnualInterestRate($annualInterestRate) {
        $this->annualInterestRate = $annualInterestRate;
    }

    public function getNumberOfYears() {
        return $this->numberOfYears;
    }

    public function setNumberOfYears($numberOfYears) {
        $this->numberOfYears = $numberOfYears;
    }

    public function getLoanAmount() {
        return $this->loanAmount;
    }

    public function setLoanAmount($loanAmount) {
        $this->loanAmount = $loanAmount;
    }

    public function getMonthlyPayment() {
        $monthlyInterestRate = $this->annualInterestRate / 1200;
        $monthlyPayment = $this->loanAmount * $monthlyInterestRate / (1 - (pow(1 / (1 + $monthlyInterestRate), $this->numberOfYears * 12)));
        return $monthlyPayment;
    }

    public function getTotalPayment() {
        $totalPayment = $this->getMonthlyPayment() * $this->numberOfYears * 12;
        return $totalPayment;
    }

}

File: Degree/P5Q1234/src/tcp/Loan.java
package tcp;

import java.io.Serializable;

public class Loan implements Serializable {

    private double annualInterestRate;
    private int numberOfYears;
    private double loanAmount;

    public Loan() {
    }

    public Loan(double annualInterestRate, int numberOfYears, double loanAmount) {
        this.annualInterestRate = annualInterestRate;
        this.numberOfYears = numberOfYears;
        this.loanAmount = loanAmount;
    }

    public double getAnnualInterestRate() {
        return annualInterestRate;
    }

    public void setAnnualInterestRate(double annualInterestRate) {
        this.annualInterestRate = annualInterestRate;
    }

    public int getNumberOfYears() {
        return numberOfYears;
    }

    public void setNumberOfYears(int numberOfYears) {
        this.numberOfYears = numberOfYears;
    }

    public double getLoanAmount() {
        return loanAmount;
    }

    public void setLoanAmount(double loanAmount) {
        this.loanAmount = loanAmount;
    }

    public double getMonthlyPayment() {
        double monthlyInterestRate = annualInterestRate / 1200;
        double monthlyPayment = loanAmount * monthlyInterestRate / (1
                - (Math.pow(1 / (1 + monthlyInterestRate), numberOfYears * 12)));
        return monthlyPayment;
    }

    public double getTotalPayment() {
        double totalPayment = getMonthlyPayment() * numberOfYears * 12;
        return totalPayment;
    }

    @Override
    public String toString() {
        return "\n\nLoan Amount : RM "+this.getLoanAmount()+
                "\nAnnual Interest Rate : "+this.getAnnualInterestRate()+"%"+
                "\nNumber Of Years : "+this.getNumberOfYears()+
                "\nMonthly Payment : RM "+this.getMonthlyPayment()+
                "\nTotal Payment : RM "+this.getTotalPayment();
    }

}
